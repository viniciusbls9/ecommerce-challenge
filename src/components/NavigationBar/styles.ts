import styled, { css } from 'styled-components'
import media from 'styled-media-query'

export const NavigationWrapper = styled.section`
  ${({ theme }) => css`
    width: 100%;
    box-shadow: 0px -3px 10px #00000029;
    background-color: ${theme.colors.white};
    height: ${theme.spacings.xxlarge};

    position: absolute;
    bottom: 0;
    z-index: 10;

    ${media.greaterThan('medium')`
      position: initial;
    `}
  `}
`

interface NavigationContainerProps {
  repeatCount: number
}

export const NavigationContainer = styled.div<NavigationContainerProps>`
  ${({ theme, repeatCount }) => css`
    width: 100%;
    max-width: ${theme.grid.container};
    height: 100%;
    margin: 0 auto;
    padding: 10px;
    display: grid;
    grid-template-columns: repeat(${repeatCount}, 1fr);
    gap: 10px;
    align-items: center;

    ${media.greaterThan('large')`
      gap: 100px;
    `}
  `}
`

interface SingleNavigationContainerProps {
  isActive: boolean
}

export const SingleNavigationContainer = styled.a<SingleNavigationContainerProps>`
  ${({ theme, isActive }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    text-align: center;
    color: ${isActive ? `${theme.colors.primary}` : `${theme.colors.gray}`};
    cursor: pointer;

    ${media.greaterThan('medium')`
      flex-direction: row;
      width: fit-content;
      justify-content: space-between;
    `}
  `}
`
export const NavigationIconImg = styled.img`
  width: 18px;
  height: 18px;

  ${media.greaterThan('medium')`
    width: 22px;
    height: 22px;
    margin-right: 15px;
  `}
`
export const NavigationName = styled.h5`
  ${({ theme }) => css`
    font-family: ${theme.font.family};
    font-size: ${theme.font.sizes.xxsmall};

    ${media.greaterThan('medium')`
      font-size: ${theme.font.sizes.xsmall};
    `}
  `}
`
