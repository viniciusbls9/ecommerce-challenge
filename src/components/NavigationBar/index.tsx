import * as S from './styles'

const NavigationBar = () => (
  <S.Wrapper>
    <h1>NavigationBar</h1>
  </S.Wrapper>
)

export default NavigationBar
