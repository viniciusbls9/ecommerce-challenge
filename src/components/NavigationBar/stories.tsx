import { Story, Meta } from '@storybook/react/types-6-0'
import NavigationBar from '.'

export default {
  title: 'NavigationBar',
  component: NavigationBar
} as Meta

export const Default: Story = () => <NavigationBar />
