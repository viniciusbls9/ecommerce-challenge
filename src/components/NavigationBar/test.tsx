import NavigationBar from 'components/NavigationBar'
import { renderWithTheme } from 'utils/helpers'

jest.mock('next/router', () => ({
  useRouter: () => ({
    asPath: '/'
  })
}))

describe('<NavigationBar />', () => {
  it('should render the NavigationBar', () => {
    const { container } = renderWithTheme(<NavigationBar />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
