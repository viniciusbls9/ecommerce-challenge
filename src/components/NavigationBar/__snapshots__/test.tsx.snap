// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> should render the NavigationBar 1`] = `
.c0 {
  grid-area: navigationBar;
  width: 100%;
  box-shadow: 0px -3px 1rem #00000029;
  background-color: #FAFAFA;
  height: 5.6rem;
  position: absolute;
  bottom: 0;
  z-index: 10;
}

.c1 {
  width: 100%;
  max-width: 104rem;
  height: 100%;
  margin: 0 auto;
  padding: 1rem;
  display: grid;
  grid-template-columns: repeat(5,1fr);
  gap: 1rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  color: #F231A5;
  cursor: pointer;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
  color: #8F8F8F;
  cursor: pointer;
}

.c3 {
  width: 1.8rem;
  height: 1.8rem;
}

.c4 {
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  font-size: 0.8rem;
}

@media (min-width:768px) {
  .c0 {
    position: initial;
  }
}

@media (min-width:1170px) {
  .c1 {
    gap: 10rem;
  }
}

@media (min-width:768px) {
  .c2 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: -webkit-fit-content;
    width: -moz-fit-content;
    width: fit-content;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (min-width:768px) {
  .c3 {
    width: 2.2rem;
    height: 2.2rem;
    margin-right: 1.5rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.2rem;
  }
}

<section
  class="c0"
>
  <div
    class="c1"
    repeatcount="5"
  >
    <a
      class="c2"
    >
      <img
        class="c3"
        src="images/inicio.svg"
      />
      <h5
        class="c4"
      >
        início
      </h5>
    </a>
    <a
      class="c5"
    >
      <img
        class="c3"
        src="images/roupas.svg"
      />
      <h5
        class="c4"
      >
        roupas
      </h5>
    </a>
    <a
      class="c5"
    >
      <img
        class="c3"
        src="images/eletronicos.svg"
      />
      <h5
        class="c4"
      >
        eletrônicos
      </h5>
    </a>
    <a
      class="c5"
    >
      <img
        class="c3"
        src="images/acessorios.svg"
      />
      <h5
        class="c4"
      >
        acessórios
      </h5>
    </a>
    <a
      class="c5"
    >
      <img
        class="c3"
        src="images/acessorios.svg"
      />
      <h5
        class="c4"
      >
        meu carrinho
      </h5>
    </a>
  </div>
</section>
`
